src/api/daemon_info.go

```diff
@@ -0,0 +1,49 @@
+package api
+
+import (
+	"errors"
+	"os/exec"
+	"strings"
+)
+
+const systemd = "systemctl"
+
+type Daemon struct {
+	name      string
+	isActive  string
+	isEnabled string
+}
+
+func execSystemctl(command string, daemon string) (string, error) {
+	cmd := exec.Command(systemd, command, daemon)
+	stdout, err := cmd.Output()
+	toReturn := strings.TrimSpace(string(stdout))
+	return toReturn, err
+}
+
+func GetSystemdDaemon(daemonName string) (Daemon, error) {
+	isActive, err := execSystemctl("is-active", daemonName)
+	if err != nil {
+		return Daemon{}, err
+	}
+
+	isEnabled, err := execSystemctl("is-enabled", daemonName)
+	if err != nil {
+		return Daemon{}, err
+	}
+
+	daemonInfo := Daemon{
+		name:      daemonName,
+		isActive:  isActive,
+		isEnabled: isEnabled,
+	}
+
+	return daemonInfo, nil
+}
+
+func GetDaemon(daemonName string, initSystem string) (Daemon, error) {
+	if initSystem == systemd {
+		return GetSystemdDaemon(daemonName)
+	}
+	return Daemon{}, errors.New("none known init system has been chosen")
```

Comment:
there should be unknown not "none known"

Reply:
ok, I will change it to no known